### DISCUSSION TITLE
Give me an exemple about AsyncCommand?

### DISCUSSION THREAD

**Discussion by antoniomldev** [spectreconsole/spectre.console#1350] [Q&A] [answered]
Hey guys, i am very newbie in this lib and C# world. I have prior knowledge in JS and so, but i am little bit confused when i tried to create async command that need to other command to be solved, anyone can be provide-me an example that what i need to do? Thanks!

**Comment by JKamsker** [comment, marked as answer]
Extract common functionality into a service class and inject that into both commmands.

Alternatively you can inject an ICommandApp and run subcommands through that
